core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -135,6 +141,11 @@ public static StructLikeSet toEqualitySet(
     }
   }
 
+  @VisibleForTesting
+  static void resetDeletePosThreadPool() {
+    deletePosThreadPool = ThreadPools.newDeleteWorkerPool();
```

Comment:
I think it is better to allow tests to pass in a custom pool that is used and cleaned up rather than adding something that replaces the static pool.